cornac/models/ann/recom_ann_hnswlib.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023 The Cornac Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+
+
+import multiprocessing
+import numpy as np
+
+from .recom_ann_base import BaseANN
+
+
+class HNSWLibANN(BaseANN):
+    """Approximate Nearest Neighbor Search with HNSWLib (https://github.com/nmslib/hnswlib/).
+
+    Parameters
+    ----------------
+    recom: object: :obj:`cornac.models.Recommender`, required
+        Trained recommender model which to get user/item vectors from.
+
+    M: int, optional, default: 16
+        Parameter that defines the maximum number of outgoing connections in the HNSW graph.
+        Higher M leads to higher accuracy/run_time at fixed ef/efConstruction
+
+    ef_construction: int, optional, default: 100
+        Parameter that controls speed/accuracy trade-off during the index construction.
+
+    ef: int, optional, default: 50
```

Comment:
We can specify here that this parameter should always be greater than K in a top-K recommendation list

Reply:
added